src/main/java/frc/robot/OI.java

```diff
@@ -44,39 +85,96 @@ public static Supplier<Double> getRotationSupplier(){
+            return new JoystickButton(kJoystick, kAlignForwardButton.getButtonID());
         }
 
         public static JoystickButton getAlignBackButton(){
-            return new JoystickButton(kJoystick, kAlignBackwardButtonID);
+            kAlignBackwardButton.setButtonName("Align Backward");
+            return new JoystickButton(kJoystick, kAlignBackwardButton.getButtonID());
         }
 
         public static JoystickButton getIntakeButton(){
-            return new JoystickButton(kJoystick, kIntakeButtonID);
+            kIntakeButton.setButtonName("Intake Button");
+            return new JoystickButton(kJoystick, kIntakeButton.getButtonID());
         }
         
         public static JoystickButton getOuttakeButton(){
-            return new JoystickButton(kJoystick, kOuttakeButtonID);
+            kOuttakeButton.setButtonName("Outtake Button");
+            return new JoystickButton(kJoystick, kOuttakeButton.getButtonID());
         }
     }
 
     public static final class Operator{
+        public enum OperatorButtons {
+            X (1, null),
+            A (2, null),
+            B (3, null),
+            Y (4, null),
+            LB (5, null),
+            RB (6, null),
+            LT (7, null),
+            RT (8, null),
+            Back (9, null),
+            Start (10, null),
+            LJ (11, null),
+            RJ (12, null);
+            
+            private final int buttonID; 
+            private String buttonName;
+          
+            OperatorButtons(int ID, String N) {
+              this.buttonID = ID;
+              this.buttonName = N;
+            }
+          
+            private int getButtonID(){
+              return this.buttonID;
+            };
+
+            private String getButtonName(){
+                return this.buttonName;
+            }
+
+            private void setButtonName(String name){
+                this.buttonName = name;
+            }
+        };
+    }
+
+    public static void putControllerButtons(){
+        ShuffleboardLayout sbDriverButtons = Shuffleboard.getTab("Controller Buttons")
+        .getLayout("Driver Buttons", BuiltInLayouts.kList)
+        .withSize(2, 5)
+        .withProperties(Map.of("Label position", "HIDDEN")); // hide labels for Variables;
+
+        ShuffleboardLayout sbOperatorButtons = Shuffleboard.getTab("Controller Buttons")
+        .getLayout("Operator Buttons", BuiltInLayouts.kList)
+        .withSize(2, 5)
+        .withProperties(Map.of("Label position", "HIDDEN")); // hide labels for Variables;
+
+        for (Driver.DriverButtons button : Driver.DriverButtons.values()) 
+        sbDriverButtons.add(String.valueOf(button.getButtonID()), "Button " + button.toString() + ": " + button.getButtonName());
 
+        for (Operator.OperatorButtons button : Operator.OperatorButtons.values()) 
+        sbOperatorButtons.add(String.valueOf(button.getButtonID()+12), "Button " + button.toString() + ": " + button.getButtonName());
```

Comment:
Use Operator.OperatorButtons.values().length instead of adding 12
Why do we need to add 12, though? Is it to prevent duplicate keys?

Reply:
Yes, that is for the titles, but the titles are hidden but still need to be different